### ISSUE TITLE
I built the tool using pyinstaller=5.13.2. i use yolov8 and ultralytics lib. pyqt5

### ISSUE THREAD

**Issue by thinhemb** [ultralytics/ultralytics#7760]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

I built the tool using pyinstaller=5.13.2. i use yolov8 and ultralytics lib. When running the exe file, the interface is double. And running regular code is fine. I discovered that when debugging, I discovered that when the model was inferred, the same thing happened

pyinstaller --noconfirm --onedir --windowed --clean --add-data "C:/Users/KNT21818/AppData/Roaming/Python/Python39/site-packages/ultralytics;ultralytics/" --add-data "C:/Users/KNT21818/.conda/envs/Ocr/Lib/site-packages/paddleocr;paddleocr/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/OCR_tool/mklml_win_2018.0.3.20180406/lib/mklml.dll;." --hidden-import "imgaug" --hidden-import "pyclipper" --hidden-import "imghdr" --hidden-import "shapely" --hidden-import "skimage" --hidden-import "onnxruntime" --hidden-import "unidecode" --hidden-import "pandas" --hidden-import "torch" --hidden-import "cv2" --hidden-import "PIL" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/common_utils;common_utils/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/config;config/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/data;data/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/log;log/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/mapper;mapper/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/object_models;object_models/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/Ocr;Ocr/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/pipeline;pipeline/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/Templates;Templates/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/yolo_detector;yolo_detector/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/init.py;." --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/dconfig.py;." --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/extract_info.py;." --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/log.py;." --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/envs;envs/" --add-data "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/Templates/WindowMain_.ui;." --collect-all "ultralytics" "C:/Users/KNT21818/Documents/WorkSpace/Tool_hyouka_consult_v1/app/main.py"

i tried the onnx model. but still the same

### Environment

Ultralytics=8.1.5, pyqt5, python3.9 , opencv=3.5

### Minimal Reproducible Example

  class YoloRegionTextDetector(BaseDetector): 

      def init(self, config): self.config = config 
          self.yolo_model_path = self.choose_model_path() 
          self.device = self.config["devices"]
          self.num_torch_threads = self.config["num_torch_threads"] 
          self.model = YOLO(self.yolo_model_path, task='detect')
          self.conf = self.config["confidence_threshold"] 
          self.iou = self.config["iou_threshold"] self.names = self.model.names
  
      def choose_model_path(self):
          if self.config["yolo_model_type"] == 'onnx':
              yolo_model_path = os.path.join(self.config["model_dir"], "yolo", "best.onnx")
          else:
              yolo_model_path = os.path.join(self.config["model_dir"], "yolo", "best.pt")
          return yolo_model_path
      
      def inference(self, img):
          results = self.model.predict(img, imgsz=[640, 640], conf=self.conf, iou=self.iou, device=self.device)
          df_result = None
          if results:
              result = results[0]
              # print(result)
              result = result.cpu()
              result = result.numpy()
              df_result = results_to_dataframe(result, self.names)
          return df_result 

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@thinhemb hello! It seems you're encountering an issue with a duplicated interface when running your packaged application with PyInstaller, but not when running the regular code. This could be related to how PyInstaller bundles the application and its dependencies, rather than an issue with the YOLOv8 model itself.

Since the problem occurs during the inference call, it might be related to the threading or GPU usage within the packaged environment. Here are a few steps you could consider:

1. Ensure that all necessary data files and dependencies are correctly included in the PyInstaller spec file.
2. Verify that the environment within the packaged application is consistent with your development environment, particularly regarding library versions and available hardware resources.
3. Check if there are any conflicts with the GUI event loop from PyQt5 and the inference process, which might cause the interface duplication.
4. Consider isolating the inference call to see if the issue persists without the GUI components.

If the problem continues, please provide more details about the duplication behavior and any error messages or logs that could help diagnose the issue further. Your willingness to submit a PR is much appreciated, and any additional findings you can share will be valuable to the community. For more detailed guidance on the predict mode and troubleshooting, you can refer to our documentation at https://docs.ultralytics.com.

**Comment by thinhemb** [comment]
I found a way to fix it by freezing the stream similar to #3997 . but the program will take a little more time the first time it runs. add multiprocessing.freeze_support()
exmp: 
```base
from PyQt5.QtWidgets import QApplication
import sys

from Templates.window_main import WindowsTool
import multiprocessing


def main():
    apps = QApplication(sys.argv)
    window = WindowsTool()
    window.show()
    sys.exit(apps.exec_())


if __name__ == "__main__":
    multiprocessing.freeze_support()
    main()

```

**Comment by glenn-jocher** [comment]
@thinhemb great to hear you've found a solution by adding `multiprocessing.freeze_support()` to your application, as mentioned in issue #3997. It's not uncommon for the first run to take a bit longer due to initial setup processes. If this workaround is acceptable for your use case and the additional startup time is manageable, it sounds like a good approach. Keep in mind that any changes in the multiprocessing behavior can affect performance, so it's always a good idea to monitor and profile your application to ensure it meets your requirements. Happy coding! 😊👍
